### PR TITLE
Ensure that the lastModifiedDate is non-null by using createdDate when null

### PR DESCRIPTION
**Pull request by dbernstein**
**Address NPE when last modified date is null.**
* * *

**JIRA Ticket**:  https://jira.lyrasis.org/browse/FCREPO-3607

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)
https://outlook.office.com/mail/search/id/AAQkADM4NjkwNGM0LWM5MzEtNDQwNC04ZGFjLTExZTEwY2EzOGRiZAAQAIYRY1x9hS9BsGzHp9CWxVs%3D

# What does this Pull Request do?
Sets lastModifiedDate to createdDate when lastModified is not available.

# How should this be tested?
1.  Export 5.1.1 content.
2.  Remove the last modified date from the on disc representation
3. Upgrade from 5 -> 6.

# Additional Notes:

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers

### CODE REVIEW

**Comment by pwinckles** [comment]
I don't disagree that this would take care of the NPE, but it feels like it's ignoring the root cause. If Fedora has always had a `lastModifiedDate` why is it null here?

**Comment by pwinckles** [comment]
1. There should be a null check on line 163
2. There should probably be WARNING logging when these fields are missing
3. There should be a test that exercises this behavior

**Comment by dbernstein** [comment]
@pwinckles :  re the null check at line 163: is it your understanding we want to the last modified date in the resource headers file and the ocfl version date to match in the case where the last modified date is missing from the RDF?

**Comment by dbernstein** [comment]
@pwinckles : this one should be good to go.

File: src/main/java/org/fcrepo/upgrade/utils/f6/ResourceMigrator.java [hunk 1]
```diff
@@ -160,8 +162,12 @@ public void close() {
...
 
         final var rdf = readRdf(info.getOuterDirectory().resolve(rdfFile(info.getNameEncoded())));
-        final var currentUpdate = RdfUtil.getDateValue(RdfConstants.FEDORA_LAST_MODIFIED_DATE, rdf);
-
+        var currentUpdate = RdfUtil.getDateValue(RdfConstants.FEDORA_LAST_MODIFIED_DATE, rdf);
+        if(currentUpdate == null) {
+            currentUpdate = Instant.now();
+            LOGGER.warn("Last modified date is not set in RDF in {}: using current time ({})",
+                        info.getFullId(), currentUpdate);
+        }
```
Comment (pwinckles) [inline src/main/java/org/fcrepo/upgrade/utils/f6/ResourceMigrator.java:170]: Maybe this defaulting logic could be extracted into a helper method.

**Review by pwinckles — COMMENTED** [review]
Looks good. Extracting a method would clean the code up a little.

**Review by pwinckles — APPROVED** [review]
👍

**Pull request merged**